### PR TITLE
combination wizard

### PR DESCRIPTION
**Pull request by matchristmann**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Tests
- [ ] Attached examples

**Test Configuration**:
* RFEM / RSTAB version:
* Python version:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: RFEM/LoadCasesAndCombinations/combinationWizard.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        if name:
+            clientObject.user_defined_name_enabled = True
+            clientObject.name = name
+
+        #What to generate
+
+        clientObject.generate_combinations = generate_combinations
+
+        if generate_combinations == 'GENERATE_LOAD_COMBINATIONS':
+
```
Comment (dogukankaratas) [inline RFEM/LoadCasesAndCombinations/combinationWizard.py:46]: Please create enumeration items in enums.py and get the selection via this class.

image

File: RFEM/LoadCasesAndCombinations/combinationWizard.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
+from RFEM.initModel import Model, clearAttributes, GetAddonStatus
+from RFEM.enums import DefinitionType, AddOn
+
+class CombinationWizard():
+    def __init__(self,
+                no: int = 1,
+                name: str = 'Load Wizard 1',
```
Comment (dogukankaratas) [inline RFEM/LoadCasesAndCombinations/combinationWizard.py:7]: It'd be nice if we can be consistent with other classes. Please use
`name: str = ''`

File: RFEM/LoadCasesAndCombinations/combinationWizard.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+        #Clear all attributes
+        clearAttributes(clientObject)
+
+        #Set Combination Wizard no.
+        clientObject.no = no
+
+        #Combination Wizard name
+        if name:
+            clientObject.user_defined_name_enabled = True
```
Comment (dogukankaratas) [inline RFEM/LoadCasesAndCombinations/combinationWizard.py:38]: Please use the following if statement here.
```
if name:
    clientObject.user_defined_name_enabled = True
    clientObject.name = name
```

File: RFEM/LoadCasesAndCombinations/combinationWizard.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        if params:
+            for key in params:
+                clientObject[key] = params[key]
+
+        model.clientModel.service.set_combination_wizard(clientObject)
+
+
+    #Setting static analysis settings
+    @staticmethod
+    def StaticAnalysisSettings(
```
Comment (dogukankaratas) [inline RFEM/LoadCasesAndCombinations/combinationWizard.py:64]: Static methods needs to cover all the instances, which init function has.

File: RFEM/LoadCasesAndCombinations/combinationWizard.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+        """
+        Args:
+            no (int): combination wizard to be edited
+            static_analysis_settings (int): Analysis Settings requiered to do combinations
+            model (RFEM Class, optional): Model to be edited
+        """
+
+        clientObject = model.clientModel.factory.create('ns0:combination_wizard')
+        clearAttributes(clientObject)
```
Comment (dogukankaratas) [inline RFEM/LoadCasesAndCombinations/combinationWizard.py:77]: Please add explanatory comments.

**Review by OndraMichal — CHANGES_REQUESTED** [review]

**Review by OndraMichal — APPROVED** [review]

**Pull request merged**